src/contacts/tests/constants.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Example constants used in tests
+"""
+from Crypto.PublicKey import RSA
+import jwt
+
+
+def generate_rsa_key():
+    """Generate priv,pub key pair for test"""
+    key = RSA.generate(2048)
+    private_key = key.export_key()
+    public_key = key.publickey().export_key()
+    return private_key, public_key
+
+
+EXAMPLE_PRIVATE_KEY, EXAMPLE_PUBLIC_KEY = generate_rsa_key()
+
+EXAMPLE_USER = "foo"
+
+EXAMPLE_USER_PAYLOAD = {"user": EXAMPLE_USER, "acct": "1234512345"}
+
+EXAMPLE_TOKEN = jwt.encode(
+    EXAMPLE_USER_PAYLOAD, EXAMPLE_PRIVATE_KEY, algorithm="RS256"
+)
+
+EXAMPLE_HEADERS = {
+    "Authorization": EXAMPLE_TOKEN,
+    "content-type": "application/json",
+}
+
+EXAMPLE_CONTACT_DB_OBJ = {
+    "username": "jdoe",
+    "label": "foo",
+    "account_num": "1234567890",
+    "routing_num": "123456789",
+    "is_external": False,
+}
+
+EXAMPLE_CONTACT = {
+    "label": "jdoe",
+    "account_num": "1234567890",
+    "routing_num": "123456789",
+    "is_external": False,
+}
+
+# Account numbers must be 10 digit numbers
+INVALID_ACCOUNT_NUMS = [
```

Comment:
might be worth adding "" and None to this list

Reply:
Looks like None hits a snag
should I add a null check and raise a warning?
```sh
env/lib/python3.8/site-packages/flask/app.py:1936: in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
contacts.py:109: in add_contact
    _validate_new_contact(req)
contacts.py:143: in _validate_new_contact
    if not re.match(r"\A[0-9]{10}\Z", req["account_num"]):
```

Reply:
Also this is the json dump that throws the error.
```json
{"label": "jdoe", "account_num": null, "routing_num": "123456789", "is_external": false}
```

Reply:
Might be worth addressing as an edge case! It looks like you could even just do:

`if not re.match(r"\A[0-9]{10}\Z", req.get("account_num", "")):`

Reply:
I ended up using
`if req["account_num"] is None or not re.match(r"\A[0-9]{10}\Z", req["account_num"]):`
as `account_num` key is present in `req` dict but has a `None` value.

Reply:
Right, good point. That looks good to me!